code/game/objects/items/devices/traitordevices.dm

```diff
@@ -80,6 +80,10 @@ effective or pretty fucking useless.
 		..()
 	if(!irradiate)
 		return
+	var/mob/living/carbon/human/human_target = M
+	if(istype(human_target) && !used && SSradiation.wearing_rad_protected_clothing(human_target)) //intentionally not checking for TRAIT_RADIMMUNE here so that tatortot can still fuck up and waste their cooldown.
```

Comment:
I think that this should be done by the radiation pulse and not by the objects doing the pulse

Reply:
not totally true, otherwise it would pulse everything on the tile

Reply:
in fact no this is definitely wrong because the microlaser doesnt ACTUALLY irradiate, because of how irradiation works now

Reply:
Correct, the laser isn't actually meant to harm/ACTUALLY irradiate someone - it's essentially just a fancy sleepypen.